File: samples/cpp/hello.cpp
#include <iostream>
#include <string>

namespace greeting {

class Greeter {
public:
    explicit Greeter(std::string name) : name_(std::move(name)) {}

    void greet() const {
        std::cout << "Hello, " << name_ << "!" << std::endl;
    }

private:
    std::string name_;
};

}  // namespace greeting

int main() {
    greeting::Greeter greeter("world");
    greeter.greet();
    return 0;
}

File: samples/haskell/Hello.hs
{-# LANGUAGE OverloadedStrings #-}
module Main (main) where

import qualified Data.Text as T
import qualified Data.Text.IO as TIO

greeting :: T.Text -> T.Text
greeting name = T.concat ["Hello, ", name, "!"]

main :: IO ()
main = TIO.putStrLn (greeting "World")
